examples/binding-scala/js/src/main/scala/com/thoughtworks/todo/Main.scala

```diff
@@ -0,0 +1,148 @@
+import org.scalajs.dom._
+import org.scalajs.dom.ext._
+import org.scalajs.dom.raw._
+import upickle.default._
+
+@JSExport object Main {
+
+  final class Todo(val title: String, val completed: Boolean)
+
+  object Todo {
+    def apply(title: String, completed: Boolean) = new Todo(title, completed)
+    def unapply(todo: Todo) = Option((todo.title, todo.completed))
+  }
+
+  final case class TodoList(text: String, hash: String, items: BindingSeq[Todo])
+
+  object Models {
+    def load() = LocalStorage("todos-binding.scala") match {
+      case None => Seq()
+      case Some(json) => read[Seq[Todo]](json)
+    }
+    def save(todos: Seq[Todo]) = LocalStorage("todos-binding.scala") = write(todos)
+
+    val allTodos = Vars[Todo](load(): _*)
+
+    val localStorageMountPoint = new MultiMountPoint(allTodos) {
+      override def set(newValue: Seq[Todo]) = save(newValue)
+      override def splice(oldSeq: Seq[Todo], from: Int, that: GenSeq[Todo], replaced: Int) = {
+        save(oldSeq.view(0, from) ++ that ++ oldSeq.view(from + replaced, oldSeq.length))
+      }
+    }
+    localStorageMountPoint.watch()
+
+    val editingTodo = Var[Option[Todo]](None)
+
+    val all = TodoList("All", "#/", allTodos)
+    val active = TodoList("Active", "#/active", for {todo <- allTodos if !todo.completed} yield todo)
+    val completed = TodoList("Completed", "#/completed", for {todo <- allTodos if todo.completed} yield todo)
+    val todoLists = Seq(all, active, completed)
+
+    def getCurrentTodoList = todoLists.find(_.hash == location.hash).getOrElse(all)
+    val currentTodoList = Var(getCurrentTodoList)
+    @dom val hashBinding = location.hash = currentTodoList.each.hash
+    hashBinding.watch()
+    window.onhashchange = { _: Any => currentTodoList := getCurrentTodoList }
+  }
+  import Models._
+
+  @dom def header = {
+    <header className="header">
+      <h1>todos</h1>
+      <input className="new-todo" autofocus={true} placeholder="What needs to be done?" onkeydown={event: KeyboardEvent =>
+      event.keyCode match {
+        case KeyCode.Enter =>
+          dom.currentTarget[HTMLInputElement].value.trim match {
+            case "" =>
+            case title =>
+              allTodos.get += new Todo(title, false)
+              dom.currentTarget[HTMLInputElement].value = ""
+          }
+        case _ =>
+        }
+      }/>
+    </header>
+  }
+
+  @dom def todoListItem(todo: Todo) = {
+    val suppressOnBlur = Var(false)
+    def submit(newTitle: String): Unit = {
+      suppressOnBlur := true
+      editingTodo := None
+      newTitle.trim match {
+        case "" =>
+          allTodos.get.remove(allTodos.get.indexOf(todo))
+        case trimmedTitle =>
+          allTodos.get(allTodos.get.indexOf(todo)) = new Todo(trimmedTitle, todo.completed)
+      }
+    }
+    val edit = <input className="edit" value={ todo.title } onblur={ 
+      if (suppressOnBlur.each) { _: Any =>
+    } else { _: Any =>
```

Comment:
Fixed in https://github.com/ThoughtWorksInc/todo/commit/f4077b5c0e1382c1461b37e66ae511d0deed997c

`onblur` is not only triggered by user interaction, but also triggered by programmatic DOM changes.
In order to suppress programmatic triggered `onblur`, we have to replace the `onblur` event listener to a dummy handler before programmatic DOM changes.